code/datums/spells/timelost_magic.dm

```diff
@@ -0,0 +1,142 @@
+/obj/effect/proc_holder/spell/aoe_turf/conjure/lostreflection //used only by Timelost spellcasters.
+	name = "Timelost Reflection"
+	desc = "Summon several reflections of your past to attack your enmies. Will also attack you if you aren't lost in time."
+	charge_max = 400
+	cast_sound = "sound/magic/CastSummon.ogg"
+
+	summon_type = list(/mob/living/simple_animal/hostile/boss/timelost_reflection)
+	summon_amt = 4
+	summon_lifespan = 300 //aka you can't have more than 4 at any time without cooldown reductions
+	range = 2
+	summon_ignore_prev_spawn_points = 1
+
+/obj/effect/proc_holder/spell/aoe_turf/conjure/forceofwill //used only by Timelost spellcasters.
+	name = "Force of Will"
+	desc = "Summon a total of 25 forcewalls randomly scattered troughout your vision range."
+	charge_max = 400
+	cast_sound = "sound/magic/CastSummon.ogg"
+
+	summon_type = list(/obj/effect/forcefield)
+	summon_amt = 25
+	summon_lifespan = 250
+	range = 7
+	summon_ignore_prev_spawn_points = 1
+
+/obj/effect/proc_holder/spell/aoe_turf/conjure/timebomb //used only by Timelost spellcasters.
+	name = "Time Bomb"
+	desc = "Summon a time bombs within your vision range that explodes after a short delay, then summons another one at a random location which does the exact same thing! Repeats itself 6 times."
+	charge_max = 600
+	cast_sound = "sound/magic/WandODeath.ogg"
+
+	summon_type = list(/obj/effect/timebomb)
+	summon_amt = 6
+	summon_ignore_prev_spawn_points = 1
+	range = 7
+
+/obj/effect/proc_holder/spell/self/time_shield
+	name = "Time Shield"
+	desc = "Shields yourself within time. Prevents you from taking any damage. Doesn't help against stuns or anything else."
+	human_req = 1
+	clothes_req = 0
+	charge_max = 600
+	cooldown_min = 50
+	invocation = "ONLY I CAN SHAPE TIME!"
+	invocation_type = "shouts"
+	school = "restoration"
+	sound = 'sound/magic/Staff_Healing.ogg'
+
+/obj/effect/proc_holder/spell/self/time_shield/cast(mob/living/carbon/human/user)
+	user.visible_message("<span class='warning'>[user] fades into time and becomes immune to all damage!</span>", "<span class='notice'>You shield yourself within time!</span>")
+	user.status_flags ^= GODMODE
+	user.icon_state = "lost_spellcaster_shielded"
+	spawn (100)
+		user.icon_state = "lost_spellcaster"
+		user.visible_message("<span class='warning'>[user] fades out of time and becomes vulnerable to damage!</span>", "<span class='notice'>Your shield vanishes!</span>")
+		user.status_flags ^= GODMODE
```

Comment:
As clever as this is it might be an issue if at some other point during the game they have this flag reset to 0, and then once this time shield wears off you'll be setting them to god mode effectively permanently.

I think it's safer to do `user.status_flags = user.status_flags & ~GODMODE` as this will remove it regardless of whether or not it's set.